
package bds;
option cc_generic_services = true;

message SeekRequest {
    optional string dict_name = 2;
    repeated bytes keys = 3;
}

message SeekResult {
    optional int32 status = 1;
    optional bytes value = 2;
}

message SeekResponse {
    repeated SeekResult results = 2;
}

message LoadDictRequest {
    optional string dict_name = 2;
    optional string dict_path = 3;
}

message LoadDictResponse {
    optional int32 status = 2;
}

service DictServer {
    rpc Seek(SeekRequest) returns(SeekResponse);
    rpc LoadDict(LoadDictRequest) returns(LoadDictResponse);
}

